Review all email patterns and the editor configurations

gabrielcaires opened the pull request:
Resolves #6618 
Resolves #6601 
Resolves #6612 


## Proposed Changes
This PR is updating the email template editor to make the Gutenberg editor more similar to the rendered email, simplifying the customization processing and reducing the complexity of maintaining emails. 

It is also enabling the users to customize the background color, margin, and spacing on the templates, previously we are using some hardcoded styles directly on the email template that was overriding the user configurations.
 
**Editor Updates:**
* Hide the post title
* Update the editor background to follow the email background
* Update the Editor area (area to insert the block) to follow the same email sizing
* Some small tweaks on the pattern

In order to reach it I had to update all existent patterns because they were created considering the email template will set some values like font sizes and colors, now it is completely explicitly on the patterns. 


## Testing Instructions
* Enable the feature toggle
* Run `npm run start` to generate the CSS
* Run the email creation tool
* Go to the email template list
* Open each template, except (Lesson Available and Course Expiration)
* Check if the look like the screenshots
* Check if the preview is rendering the emails correctly.


## Screenshots
### Student Emails
* Course Completed  

  <img width="1170" alt="image" src="https://user-images.githubusercontent.com/38718/224388041-c11c9bee-e23d-4385-8fbb-fd3644c5f957.png">
  
* Message Reply Received

  <img width="1167" alt="image" src="https://user-images.githubusercontent.com/38718/224388222-7f3ac87a-9844-421c-b8a6-9247c4301ea1.png">
  
* Quiz Graded

  <img width="1166" alt="image" src="https://user-images.githubusercontent.com/38718/224388372-ba31d838-6641-44fb-a203-888a2e8fb428.png">


----

### Student Emails

* Course Assigned

  <img width="1180" alt="image" src="https://user-images.githubusercontent.com/38718/224389032-be00c41e-7675-46b8-97fb-44156282f4f7.png">

* Course Complete

  <img width="1177" alt="image" src="https://user-images.githubusercontent.com/38718/224389141-0616ca6c-2db2-43e8-b516-90c804a0a1e7.png">


* Course Created

  <img width="1175" alt="image" src="https://user-images.githubusercontent.com/38718/224389332-52b83c9d-46af-4700-9563-3f33014253c6.png">

* Course Started
  
  <img width="1171" alt="image" src="https://user-images.githubusercontent.com/38718/224389475-0f48cfa8-b370-449e-b574-b5219d4ad84a.png">


* Lesson Completed 
  <img width="1177" alt="image" src="https://user-images.githubusercontent.com/38718/224389651-e7ce6b4f-072f-4b61-9cf9-b77c91d90d1e.png">


* Message Reply Received
  <img width="1174" alt="image" src="https://user-images.githubusercontent.com/38718/224389738-ad3963cf-87f3-4e35-a14f-27a0f7f52f5c.png">

* Quiz Submitted
  <img width="1176" alt="image" src="https://user-images.githubusercontent.com/38718/224389959-ffe73471-a12d-4816-9911-ac32a57b512e.png">


* Student send message
  <img width="1174" alt="image" src="https://user-images.githubusercontent.com/38718/224390097-5a24da1d-d843-48a4-a1fd-45469f184252.png">



## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [ ] Acceptance criteria is met
- [x] Decisions are publicly documented
- [ ] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [ ] All strings are translatable (without concatenation, handles plurals)
- [ ] Data is [sanitized](https://developer.wordpress.org/apis/security/sanitizing/) / [escaped](https://developer.wordpress.org/apis/security/escaping/)
- [ ] Follows our naming conventions (P6rkRX-4oA-p2)
- [ ] Hooks (p6rkRX-1uS-p2) and functions are documented
- [ ] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [ ] New UIs match the designs
- [x] Test cases are written and passing (including feature flags)
- [ ] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [x] Code is tested on the minimum supported PHP and WordPress versions
- [ ] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [ ] We are generally happy with it and don’t feel like it immediately needs to be rewritten
- [ ] Application performance has not degraded
- [ ] "Needs Documentation" label is added if this change requires updates to documentation
- [ ] Known issues are created as new GitHub issues
- [ ] Continuous Integration build is passing

gabrielcaires committed:
Set the editor size to be equal to the email size

gabrielcaires committed:
Review course created email pattern

gabrielcaires committed:
Review course created email pattern

gabrielcaires committed:
Review course student complete lesson pattern

gabrielcaires committed:
Review teacher message reply  pattern

gabrielcaires committed:
Fix wrong pattern namespace

gabrielcaires committed:
Review the student submits quiz pattern

gabrielcaires committed:
Review the student sends message pattern

gabrielcaires committed:
Adjust top padding on the email patterns

gabrielcaires committed:
Fix the email pattern colors

gabrielcaires committed:
Merge branch 'trunk' into update/email-patterns

gabrielcaires committed:
Review the student course completed pattern

gabrielcaires committed:
Review the teacher course assigned pattern

gabrielcaires committed:
Review the quiz graded email pattern

gabrielcaires committed:
Review the tacher course complete email pattern

gabrielcaires committed:
Review the teacher's student reply message email pattern

gabrielcaires commented on includes/internal/emails/class-email-blocks.php:
```diff
@@ -40,6 +40,9 @@ class Email_Blocks {
 	public const EMAIL_THEME_SETTINGS = [
 		'version'  => 2,
 		'settings' => [
+			'layout'     => [
```
Set the editor size

gabrielcaires commented on includes/internal/emails/class-email-seeder-data.php:
```diff
@@ -82,7 +82,7 @@ public function get_email_data(): array {
 				'subject'     => __( '[student:displayname] has submitted a quiz', 'sensei-lms' ),
 				'description' => __( 'Quiz Submitted', 'sensei-lms' ),
-				'content'     => '<!-- wp:pattern {"slug":"sensei-emails/student-submits-quiz"} /-->',
+				'content'     => '<!-- wp:pattern {"slug":"sensei-lms/student-submits-quiz"} /-->',
```
Fix issue #6630

gabrielcaires commented on templates/emails/block-email-template.php:
```diff
@@ -74,54 +79,6 @@
 			padding: 40px 24px 24px;
 		}
 
-		.narrow .content {
```
Removing unused style

Imran92 reviewed (commented):
I'm testing on wp 5.9. I see the following error message when I go to the templates editor. For example, Student Emails -> Course Completed shows this message in the editor -

![image](https://user-images.githubusercontent.com/6820724/224675612-9e3faaeb-691c-4605-b056-0f89be4a3e64.png)

gabrielcaires committed:
Merge branch 'trunk' into update/email-patterns

gabrielcaires committed:
Hide email title on WP 5.9

gabrielcaires committed:
Remove the layout attribute because it is not supported by WP 5.9

gabrielcaires committed:
Fix the background editor color on wp 6.1

gabrielcaires committed:
Fix rendering issue cause by incorrect sytle property order

gabrielcaires commented:
> I'm testing on wp 5.9. I see the following error message when I go to the templates editor. For example, Student Emails -> Course Completed shows this message in the editor -
> 
> ![image](https://user-images.githubusercontent.com/6820724/224675612-9e3faaeb-691c-4605-b056-0f89be4a3e64.png)

@Imran92 

Fixed  
7ebf4dc and  3759c78

Zero ideas why it was not happening on my env.

gabrielcaires committed:
Merge branch 'trunk' into update/email-patterns

Imran92 reviewed (commented):
Thanks, it's now loading in my editor without any issue. The "Manage Students" button has a background color in the editor in the Course Started email, but doesn't have it in the sent email. I'm using the Astra theme. Not sure if there's any other email like this
![image](https://user-images.githubusercontent.com/6820724/224957717-581eb257-9580-4a25-b251-a938bc4b70a0.png)

Edit:
Same for the lesson complete email
![image](https://user-images.githubusercontent.com/6820724/224959145-536a78d5-4174-427d-ba71-741908e10efe.png)

gabrielcaires committed:
Fix the button colors

gabrielcaires committed:
Remove the border radius

gabrielcaires committed:
Fix layout incosistencies on preview

gabrielcaires committed:
Disable the theme color pallete

gabrielcaires committed:
Fix missing style

gabrielcaires commented:
> Thanks, it's now loading in my editor without any issue. The "Manage Students" button has a background color in the editor in the Course Started email, but doesn't have it in the sent email.

@Imran92 

I review all emails and fixed them.
I also fixed the wrong font family using astra, so please just run `npm run start` before testing the emails.

Imran92 committed:
Merge branch 'trunk' into update/email-patterns

Imran92 reviewed (approved):
LGTM 👍
